sklearn/linear_model/tests/test_sparse_coordinate_descent.py

```diff
@@ -374,3 +375,18 @@ def test_sparse_enet_coordinate_descent():
     )
     with pytest.warns(ConvergenceWarning, match=warning_message):
         clf.fit(X, y)
+
+
+def test_read_only_buffer():
+    """Test that sparse coordinate descent works for read-only buffers"""
+    clf = ElasticNet(alpha=0.1, copy_X=False)
+    X = sp.random(100, 20, format="csc")
+
+    def make_read_only(obj, attr):
+        curr_attr = getattr(obj, attr)
+        RO_attr = create_memmap_backed_data(curr_attr)
+        setattr(obj, attr, RO_attr)
+
+    make_read_only(X, "data")
```

Comment:
Since you only make "data" read-only, I think the helper is no longer necessary and you can write it as flat code